create_shards.c

```diff
@@ -91,12 +103,69 @@ master_create_distributed_table(PG_FUNCTION_ARGS)
 		ereport(ERROR, (errmsg("could not find column: %s", partitionColumnName)));
 	}
 
-	/* we only support hash partitioning method for now */
-	if (partitionMethod != HASH_PARTITION_TYPE)
+	partitionColumnVar = ColumnNameToColumn(distributedTableId, partitionColumnName);
+	partitionColumnTypeId = partitionColumnVar->vartype;
+
+	if (partitionMethod == HASH_PARTITION_TYPE)
+	{
+		partitionColumnOpClassId = GetDefaultOpClass(partitionColumnTypeId, HASH_AM_OID);
```

Comment:
Calling `GetDefaultOpClass` here and doing validity checking further down is confusing to me. At first I was going to suggest moving the validity checking into this `if` block, but I really think we should just move the validity checking into the `GetSupportRoutine` function, i.e. have it `ereport` directly rather than returning `InvalidOid` if there is no default op class for the specified type.